<?xml version="1.0"?>

<launch>

  <!-- URDF robot description -->
  <param name="robot_description" command="cat $(find dynamixel_hardware_interface)/urdf/test.urdf" />
  <!-- Publish robot's state in the tf tree -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="rob_state_pub" />

  <!-- Parameters for the hardware interface -->
  <param name="dynamixel_loop/loop_hz" value="10" />
  <param name="dynamixel_loop/cycle_time_error_threshold" value="100" />


  <!-- To move the dummy robot with a gui, bypassing the controller manager -->
  <!-- <param name="use_gui" value="true"/>
  <node pkg="joint_state_publisher" type="joint_state_publisher" name="dummy_robot" /> -->

  <!-- launch our hardware interface -->
  <node pkg="dynamixel_hardware_interface" type="dynamixel_control_renewed" name="dynamixel_hardware_interface" />

  <!-- Start a controller for our dummy robot -->
  <rosparam file="$(find dynamixel_hardware_interface)/config/dummy_position.yaml"/>
  <node name="controller" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="/dynamixel_loop/first_position_controller /dynamixel_loop/joint_state_controller /dynamixel_loop/second_position_controller" />
</launch>